src/general/mppequil.cpp

```diff
@@ -912,11 +912,17 @@ int main(int argc, char** argv)
                 } else if (name == "omega") {
                     mix.netProductionRates(species_values);
                 } else if (name == "Omega11") {
-                    Map<ArrayXd>(species_values,mix.nSpecies()) =
-                        (ArrayXd(mix.nSpecies()) << mix.collisionDB().Q11ee(),mix.collisionDB().Q11ii()).finished();
+                    int k = mix.hasElectrons() ? 1 : 0;
+                    if (k == 1)
+                        species_values[0] = mix.collisionDB().Q11ee();
+                    Map<ArrayXd>(species_values+k, mix.nHeavy()) =
+                        mix.collisionDB().Q11ii();
                 } else if (name == "Omega22") {
```

Comment:
You should probably release a new version.

Reply:
Hey @rubendibattista, totally agree for the refactoring suggestion.  This code is quite old since it's been around more or less from the beginning of the library where there weren't many functions yet (and my coding skills were not as good :)).  In fact, I have had in mind for a while to provide a kind of module that allows you to get mixture properties with just a string name that would do basically what you suggest.  This could be useful for a wide range of things, not just mppequil.  For example, you could then easily build a "PropertyGetter" that could be constructed with a list of property names and would return the a vector of data each time you call it with the corresponding properties.  The fix here is obviously just for this specific bug, and I think refactoring this code should be a separate issue/PR combo.

For the docopt suggestion: actually I would like to minimize the amount of third-party libs that we use if possible, and since the parsing here is rather simple, then I think this is still OK.  Perhaps a better approach would be just to port all the command line tools into python and then use argparse...  Again, a discussion for another day I think :).

I will try to release a new version this weekend.  Good idea!

Since it seems you are OK with the code for this specific fix (with nice suggestions for future work), I will go ahead and merge.

Reply:
It might also be even more useful to expose the Python API, and write the "executables" directly in Python.

I might try to give it a go, starting from the `Mixture` class.

Reply:
Yep exactly what I meant! That would be great if you want to do this.  Note that @momom4urice has started this in the past, so it might be useful to get in touch with him to see where the work stands.

Reply:
Yep exactly what I meant! That would be great if you want to do this.  Note that @momom4urice has started this in the past, so it might be useful to get in touch with him to see where the work stands.